kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -0,0 +1,392 @@
+
+    def when[S1](condition: => Boolean < S1): A < (S & S1 & Abort[Maybe.Empty]) =
+        condition.map(c => if c then effect else Abort.fail(Maybe.Empty))
+
+    def explicitThrowable(using Flat[A]): A < (S & Abort[Throwable]) =
+        Abort.catching[Throwable](effect)
+
+    def tap[S1](f: A => Any < S1): A < (S & S1) =
+        effect.map(a => f(a).as(a))
+
+    def unless[S1](condition: Boolean < S1): A < (S & S1 & Abort[Maybe.Empty]) =
+        condition.map(c => if c then Abort.fail(Maybe.Empty) else effect)
+
+end extension
+
+extension [A, S, E](effect: A < (Abort[E] & S))
+    def handleAbort(
+        using
+        ct: ClassTag[E],
+        tag: Tag[E],
+        flat: Flat[A]
+    ): Result[E, A] < S =
+        Abort.run[E](effect)
+
+    def abortToChoice(
+        using
+        ct: ClassTag[E],
+        tag: Tag[E],
+        flat: Flat[A]
+    ): A < (S & Choice) =
+        Abort.run[E](effect).map(e => Choice.get(e.fold(_ => Nil)(List(_))))
+
+    def someAbortToChoice[E1 <: E]: SomeAbortToChoiceOps[A, S, E, E1] = SomeAbortToChoiceOps(effect)
+
+    def handleSomeAbort[E1 <: E]: HandleSomeAbort[A, S, E, E1] = HandleSomeAbort(effect)
+
+    def catchAbort[A1 >: A, S1](fn: E => A1 < S1)(
+        using
+        ClassTag[E],
+        Tag[E],
+        Flat[A]
+    ): A1 < (S & S1) =
+        Abort.run[E](effect).map {
+            case Result.Fail(e)    => fn(e)
+            case Result.Panic(e)   => throw e
+            case Result.Success(v) => v
+        }
+
+    def catchAbortPartial[A1 >: A, S1](fn: PartialFunction[E, A1 < S1])(
+        using
+        ClassTag[E],
+        Tag[E],
+        Flat[A]
+    ): A1 < (S & S1 & Abort[E]) =
+        Abort.run[E](effect).map {
+            case Result.Fail(e) =>
+                if fn.isDefinedAt(e) then fn(e)
+                else Abort.fail(e)
+            case Result.Panic(e)   => throw e
+            case Result.Success(v) => v
+        }
+
+    def catchSomeAbort[E1 <: E]: CatchSomeAbort[A, S, E, E1] = CatchSomeAbort(effect)
+
+    def catchSomeAbortPartial[E1 <: E]: CatchSomeAbortPartialOps[A, S, E, E1] = CatchSomeAbortPartialOps(effect)
+
+    def swapAbort(
+        using
+        cte: ClassTag[E],
+        cta: ClassTag[A],
+        tagE: Tag[E],
+        fl: Flat[A]
+    ): E < (S & Abort[A]) =
+        val handled: Result[E, A] < S = Abort.run[E](effect)
+        handled.map((v: Result[E, A]) => Abort.get(v.swap))
+    end swapAbort
+
+    def swapSomeAbort[E1 <: E]: SwapSomeAbortOps[A, S, E, E1] = SwapSomeAbortOps(effect)
+
+    def implicitThrowable[ER](
+        using
+        ev: E => Throwable | ER,
+        f: Flat[A],
+        reduce: Reducible[Abort[ER]]
+    ): A < (S & reduce.SReduced) =
+        Abort.run[Throwable](effect.asInstanceOf[A < (Abort[Throwable | ER] & S)])
+            .map(_.fold(e => throw e.getFailure)(identity))
+end extension
+
+class SomeAbortToChoiceOps[A, S, E, E1 <: E](effect: A < (Abort[E] & S)) extends AnyVal:
```

Comment:
@jiveshungerford The new `Reducible` mechanism worked well for other APIs but it got quite complicated here. I had to add these `Ops` classes + make the methods require an additional `()`.

Reply:
I've also removed the `Options` APIs since the effect got removed.